lib/workos/authentication_response.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+# typed: true
+
+module WorkOS
+  # The AuthenticationResponse class represents a User as well as an corresponding
+  # response data that can later be appended on.
+  class AuthenticationResponse
```

Comment:
I wonder if we can just return a hash with a User key instead of a full new class? I feel like we're going to be using this return format beyond just Authentication. When the responses get more complex then we can move them to their own classes?

Reply:
I'm happy to do this either way.  I know that for the other SDKs (Node, PHP) a new class was used for the response so it would be easier to build off of in the future, but I was thinking about whether a has with a User key would be easier for now.

Reply:
After discussing with Willman, for now, since parsing is done in the class and not in the API method, we'll go forward with the AuthenticationResponse class.